code/inc/gfx/element/gfx_mesh.hpp

```diff
@@ -0,0 +1,246 @@
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**************************************************************************************************/
+
+#pragma once
+
+/*=================================================================================================
+** 1.  REFERENCES
+**===============================================================================================*/
+
+/*=================================================================================================
+** 2.  INCLUDE FILES
+**===============================================================================================*/
+
+#include "gfx/gfx_common.hpp"
+
+#include "gfx/element/gfx_element.hpp"
+#include "gfx/element/gfx_vertex.hpp"
+#include "gfx/shader/gfx_shader.hpp"
+#include "gfx/gfx_glvibuffer.hpp"
+
+/*=================================================================================================
+** 3.  DECLARATIONS
+**
+** 3.1 Macros
+**===============================================================================================*/
+
+/*=================================================================================================
+** 3.2 Types and Classes
+**===============================================================================================*/
+namespace GFX
+{
+    namespace Element
+    {
+        /**************************************************************************************************/
+        /**
+        * \par Details: 
+        */
+        class Mesh : public Element
+        {
+
+        public:
+            /*********************************/
+            // Public type definitions
+            /*********************************/
+
+            /*********************************/
+            // Public member variables
+            /*********************************/
+
+            glm::vec3 m_Color;
+            float32_t m_Roughness;
+            float32_t m_Metallic;
+
+            /*********************************/
+            // Constructors/Destructor
+            /*********************************/
+
+            /**************************************************************************************************/
+            /**
+            * \brief The default constructor for the Mesh.
+            */
+            Mesh(void) = default;
+
+            /**************************************************************************************************/
+            /**
+            * \brief The default destructor for the Mesh.
+            */
+            virtual ~Mesh(void);
+
+            /*********************************/
+            // Public functions
+            /*********************************/
+
+            /**************************************************************************************************/
+            /**
+            * \brief 
+            * 
+            * \param[in] 
+            * 
+            * \retval 
+            */
```

Comment:
Fill in function comments and for those below.

Reply:
Fixed https://github.com/tdiego1/worldweaver/pull/19/commits/1a0df967e70f1752177ebca99e4618e4ec06c26d